### PR TITLE
fix/928_added back arrow button to the solution-detail component

### LINKED ISSUE #928: Solutions UI improvement
image

instead of empty bar wasting vertical space, add a back arrow https://lucide.dev/icons/arrow-left next to the pfp inside a circle matching the size of the pfp circle.

image

optionally, put buttons to the next line to match the challenge layout with name&date on one line and buttons on the next
image

### PR DESCRIPTION
**Pull request by PrathameshSonje**
UI improvements :
Added a back arrow icon next to the profile picture  and shifted the icon to the next line

Closes #928 

image

### CODE REVIEW

File: apps/web/src/app/[locale]/challenge/[slug]/solutions/_components/solution-detail.tsx [hunk 1]
```diff
@@ -20,6 +20,7 @@ import { pinOrUnpinSolution } from './_actions';
 import { isAdminOrModerator, isAuthor } from '~/utils/auth-guards';
 import { SolutionDeleteDialog } from './delete';
 import { useParams } from 'next/navigation';
+import { ArrowLeft } from 'lucide-react';
```
Comment (odicho) [inline apps/web/src/app/[locale]/challenge/[slug]/solutions/_components/solution-detail.tsx:23]: Small nit: move this icon to @repo/ui/icons

**Comment by odicho** [comment]
Nice work!

Is the icon by default with a background or is it in a hover state in the screenshot? I think to keep it consistent with other close/back icons, can you try to use an icon that is just the arrow and then has the background on hover?

**Comment by odicho** [comment]
CC: @PickleNik I think there needs to be a tiny bit more spacing here. Thoughts?

**Comment by PickleNik** [comment]
I don't like how it looks multi-lined, didn't think of that, maybe `items-start` ?

**Comment by PickleNik** [comment]
> CC: @PickleNik I think there needs to be a tiny bit more spacing here. Thoughts?

Yeah I agree, `gap-3` instead of `gap-2`

**Comment by PickleNik** [comment]
also it looks like it's not very well formatted in some places, and idk if that `<div>` around the `<Link>` is necessary
image
Screenshot 2023-10-22 at 8 11 12 PM

**Comment by PrathameshSonje** [comment]
@odicho @PickleNik Thankyou for the feedback, I will do the changes.

**Comment by PrathameshSonje** [comment]
I did the changes changes to the arrow button

WhatsApp Image 2023-10-24 at 01 59 19_4b9b172a
WhatsApp Image 2023-10-24 at 01 59 42_d8eea8a9

**Comment by bautistaaa** [comment]
is it me or is that C not in the center

**Comment by bautistaaa** [comment]
> I did the changes to the arrow button
> 
> WhatsApp Image 2023-10-24 at 01 59 19_4b9b172a WhatsApp Image 2023-10-24 at 01 59 42_d8eea8a9

you have a conflict

**Comment by odicho** [comment]
> is it me or is that C not in the center

Isn't that just the pfp pulled in from GitHub?

**Comment by bautistaaa** [comment]
> > is it me or is that C not in the center
> 
> Isn't that just the pfp pulled in from GitHub?

nah it’s a fallback to the users initials if there’s no photo

**Comment by odicho** [comment]
> > > is it me or is that C not in the center
> > 
> > 
> > Isn't that just the pfp pulled in from GitHub?
> 
> nah it’s a fallback to the users initials if there’s no photo

Oh, hahahaha! Then yeah, it needs to be fixed! If not in this PR, let me know and I'll take care of it!

**Comment by bautistaaa** [comment]
conflicts!

**Comment by bautistaaa** [comment]
@PrathameshSonje can you fix these conflicts and ill merge

**Comment by PrathameshSonje** [comment]
> @PrathameshSonje can you fix these conflicts and ill merge

can you review the code , I have resolved the conflicts

**Comment by PickleNik** [comment]
> can you review the code , I have resolved the conflicts

could u run `pnpm checks:fix` ?

**Comment by PrathameshSonje** [comment]
> > can you review the code , I have resolved the conflicts
> 
> could u run `pnpm checks:fix` ?

Done.

**Comment by PrathameshSonje** [comment]
what does `pnpm checks:fix` do ?

**Comment by PickleNik** [comment]
> what does `pnpm checks:fix` do ?
fixes checks, I think the issue was one of the brackets wasn't formatted so the Prettier check failed
Screenshot 2023-10-26 at 9 46 04 PM

**Review by PickleNik — APPROVED** [review]

**Pull request merged**